### ISSUE TITLE
Exact hyperparameters for reproducing results

### ISSUE THREAD

**Issue by aimzer** [google-research/simclr#77]
I just want to make sure I'm using the right hyperparameters with resnet 50. Following the paper, the learning rate should be 0.1 and warm-up epochs should be 40 = 800*5%, right ?

Thank you

**Comment by chentingpc** [comment]
For hyperparameters in SimCLRv2, please see section 3.1 in https://arxiv.org/pdf/2006.10029.pdf.  Lr = 0.1 × sqrt(BatchSize), warmup is 5% of total (800 in simclrv2).

**Comment by aimzer** [comment]
Thank you !

**Comment by aimzer** [comment]
Hi again, thank you for the response.
I've trained the model with the specified config (pretraining and linear classification with resnet 50) and I obtain a top1-acc of 70.4% (compared to the reported 71.7%) and a top5-acc of 89.9%. Are there additional hyperparams to take into account ?

**Comment by chentingpc** [comment]
I previously got 71.1% (vs 71.7% with momentum contrast) in the same setup, so I think there is probably a ~0.5% difference here. I'm not sure where the difference comes from. Is it possible that training a bit longer with the linear classifier could reduce the gap?

**Comment by aimzer** [comment]
Thank you for your response ! It's not possible to use momentum contrast with this repo I think is it possible to have the code and hyperparameters for that setting please ? I will train the linear classifier for a little longer and get back with results. Thank you.

**Comment by aimzer** [comment]
@chentingpc Thank you for your help. Unfortunately, I wasn't able to close the 0.7% gap (between 70.4 and 71.1) by the classifier for longer (starting from different pretraining checkpoints). I'm using the standard imagenet dataset and tfds to unpack it so I think our setups are identical. I'd like to build on top of the results reported here so is it possible to have the momentum contrast code and hyperparameters please ?

**Comment by chentingpc** [comment]
close it for now as there's no plan for releasing the momentum encoder part (wanting to keep the code clean).

**Comment by YanchaoYang** [comment]
> Hi again, thank you for the response.
> I've trained the model with the specified config (pretraining and linear classification with resnet 50) and I obtain a top1-acc of 70.4% (compared to the reported 71.7%) and a top5-acc of 89.9%. Are there additional hyperparams to take into account ?

Acutally, how do you train the models? Is it on TPU or GPU? How many of them?
